Tests/L2Tests/L2Tests/tests/PersistentStore_L2Test_dataCorrupt.cpp

```diff
@@ -0,0 +1,183 @@
+        uint32_t status = Core::ERROR_GENERAL;
+        Core::JSONRPC::Message message;
+        ServiceMock service;
+
+        /* Activate plugin in constructor */
+        status = ActivateService("org.rdk.PersistentStore");
+        EXPECT_EQ(Core::ERROR_NONE, status);
+
+}
+
+/**
+ * @brief Destructor for PersistentStore L2 test class
+ */
+PersistentStore_L2Test::~PersistentStore_L2Test()
+{
+    TEST_LOG("************PersistentStore Destructor***********\n");
+    uint32_t status = Core::ERROR_GENERAL;
+    status = DeactivateService("org.rdk.PersistentStore");
+    EXPECT_EQ(Core::ERROR_NONE, status);
+}
+
+/********************************************************
+************Test case Details **************************
+** 1. Set key value with specified namespace
+** 2. Check for Database file Path if it exists
+** 3. Destroy the file 
+** 4. Invoke deleteNamespace and getValue 
+*******************************************************/
+
+TEST_F(PersistentStore_L2Test, PersistentStoreDataCorruption)
+{
+    JSONRPC::LinkType<Core::JSON::IElement> jsonrpc(PERSISTENTSTORE_CALLSIGN, PERSISTENTL2TEST_CALLSIGN);
+    StrictMock<AsyncHandlerMock_PStore> async_handler;
+    uint32_t status = Core::ERROR_GENERAL;
+    JsonObject params;
+    JsonObject result;
+    std::string message;
+    JsonObject expected_status;
+
+
+    status = InvokeServiceMethod("org.rdk.PersistentStore.1", "getValue", params, result);
+    EXPECT_EQ(Core::ERROR_GENERAL, status);
+    EXPECT_FALSE(result["success"].Boolean());
+
+    params["namespace"] = "ns1";
+    params["key"] = "key2";
+    params["value"] = "012";
+
+    //Sets the value of a key in the the specified namespace
+    status = InvokeServiceMethod("org.rdk.PersistentStore.1", "setValue", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_TRUE(result["success"].Boolean());
+    
+    
+    //Returns the value of a key from the specified namespace
+    status = InvokeServiceMethod("org.rdk.PersistentStore.1", "getValue", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_TRUE(result["success"].Boolean());   
+    
+    //Returns the size occupied by each namespace
+    status = InvokeServiceMethod("org.rdk.PersistentStore.1", "getStorageSize", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_TRUE(result["success"].Boolean());
+
+    //Delete the path where DB file Exist
+
+    if (Core::File(string("/opt/secure/persistent/rdkservicestore")).Exists()) {
+
+       EXPECT_TRUE(Core::File(string("/opt/secure/persistent/rdkservicestore")).Destroy());
+       
+    }
+    EXPECT_FALSE(Core::File(string("/opt/secure/persistent/rdkservicestore")).Exists());
+
+    //Returns the value of a key from the specified namespace
+    status = InvokeServiceMethod("org.rdk.PersistentStore.1", "getValue", params, result);
+    EXPECT_EQ(Core::ERROR_NONE, status);
+    EXPECT_TRUE(result["success"].Boolean());
```

Comment:
If the db file is deleted, the how does the getValue is succeed? Where is the data corruption check here?